From beb43be7d81119f2d8ea00a632563355880ca085 Mon Sep 17 00:00:00 2001
From: James Page <james.page@chainguard.dev>
Date: Tue, 12 Aug 2025 09:27:46 +0000
Subject: [PATCH] kc.sh: misc FIPS related fixes

Ensure that JAVA_TOOL_OPTIONS is cleared if it contains --module-path as
this breaks the main entrypoint to Keycloak.

Provide Bouncycastle FIPS jars to the Keycloak entrypoint using
--class-path.
---
 quarkus/dist/src/main/content/bin/kc.sh | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/quarkus/dist/src/main/content/bin/kc.sh b/quarkus/dist/src/main/content/bin/kc.sh
index 9c0b946e85..375fdc286e 100644
--- a/quarkus/dist/src/main/content/bin/kc.sh
+++ b/quarkus/dist/src/main/content/bin/kc.sh
@@ -1,5 +1,13 @@
 #!/bin/sh
 
+# --module-path confuses the main entrypoint to keycloak so clear
+# JAVA_TOOL_OPTIONS if this is found.
+if [ -n "$JAVA_TOOL_OPTIONS" ]; then
+    if echo "$JAVA_TOOL_OPTIONS" | grep -q -- "--module-path"; then
+        unset JAVA_TOOL_OPTIONS
+    fi
+fi
+
 case "$(uname)" in
     CYGWIN*)
         IS_CYGWIN="true"
@@ -145,6 +153,15 @@ else
 fi
 JAVA_OPTS="$JAVA_OPTS $JAVA_LOCALE"
 
+# Ensure Bouncycastle FIPS jars are on the Java Classpath
+BC_FIPS_DIR="/usr/share/java/bouncycastle-fips"
+if [ -d "$BC_FIPS_DIR" ]; then
+    BC_FIPS_CLASSPATH=$(find "$BC_FIPS_DIR" -type f -name "*.jar" | tr '\n' ':')
+    # Remove trailing colon if any
+    BC_FIPS_CLASSPATH=${BC_FIPS_CLASSPATH%:}
+    JAVA_OPTS="$JAVA_OPTS --class-path $BC_FIPS_CLASSPATH"
+fi
+
 if [ -n "$JAVA_OPTS_APPEND" ]; then
   echo "Appending additional Java properties to JAVA_OPTS"
   JAVA_OPTS="$JAVA_OPTS $JAVA_OPTS_APPEND"
-- 
2.50.1

